spec/performance/requests/vendor_api/applications_controller_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe VendorApi::ApplicationsController, type: :request do
+  context 'given two providers with 100 applications in various states' do
```

Comment:
Do you know how long does this test takes to run? If it's too slow we might want to split it out / not run it by default.

Reply:
It takes about 20s to run on my machine, due to the setup - I've excluded all specs under `spec/performance` from the default rake task in the latest commit. You can still run them explicitly with `rake spec:performance` (or `rspec spec/performance`) if you want to.